ArduPlane/soaring.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Plane.h"
+
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell, Samuel Tabor, Andrey Kolobov, and Iain Guilliard
+*/
+void Plane::update_soaring() {
+    
+    if (!g2.soaring_controller.is_active()) {
+        return;
+    }
+    
+    g2.soaring_controller.update_vario();
+
+    // Check for throttle suppression change.
+    switch (control_mode){
+    case AUTO:
+        g2.soaring_controller.suppress_throttle();
+        break;
+    case FLY_BY_WIRE_B:
+    case CRUISE:
```

Comment:
Right now we are doing RTL when we reach SOAR_ALT_MIN in CRUISE (this also happens if we reach it in LOITER and then exit to CRUISE) .  If we want to re-enable throttle and fly back to the target altitude we can do the same as with AUTO. @tridge  what did you have in mind?